test/ContractTestContext.ts

```diff
@@ -18,92 +18,121 @@ import type {
+};
+
+type ContractRolesWithDefaults = {
+    [contractKey: string]: Partial<DefaultRoles>;
+    defaults: DefaultRoles & ContractRoles;
+};
+
+type ContractPath = string;
+type ContractConstructorArgs = any[];
+type ContractMap = {
+    [contractKey: string]: [ContractPath, ...ContractConstructorArgs];
 }
 
-export type ContractTestContext = {
+/* Return object matching contractMap keys with deployed contract objects. contractMap values should be
+   an array where first item is path to contract and remaining items are constructor args to pass. First
+   argument to contructor will be admin address which is expected by AccessController */
+async function deployContracts <T> (contractMap: ContractMap, contractRoles: ContractRolesWithDefaults): Promise<T> {
+    const contracts = {};
+    await Promise.all(Object.entries(contractMap).map(async ([contractKey, [contractPath, ...constructorArgs]]) => {
+        const factory = await ethers.getContractFactory(contractPath);
+        const overrideRoles = contractRoles[contractPath];
+        const addressAdmin = (overrideRoles && overrideRoles.admin) || contractRoles.defaults.admin;
+        const contract = await factory.deploy(addressAdmin, ...constructorArgs);
+        contracts[contractKey] = contract;
+    }));
+    return contracts as T;
+}
+
+type ArchContracts = {
+    cdp: CDPosition;
     coordinator: Coordinator;
+    exchanger: Exchanger;
+    leverageAllocator: LeverageAllocator;
+    leverageEngine: LeverageEngine;
+    lvUSD: LvUSDToken;
+    parameterStore: ParameterStore;
+    positionToken: PositionToken;
+    vault: VaultOUSD;
+};
+
+export type ContractTestContext = ArchContracts & {
     owner: SignerWithAddress;
     addr1: SignerWithAddress;
     addr2: SignerWithAddress;
     addr3: SignerWithAddress;
     treasurySigner: SignerWithAddress;
-    // Archimedes deployed contract instances
-    cdp: CDPosition;
-    vault: VaultOUSD;
-    lvUSD: LvUSDToken;
-    exchanger: Exchanger;
-    leverageEngine: LeverageEngine;
-    leverageAllocator: LeverageAllocator;
-    positionToken: PositionToken;
-    parameterStore: ParameterStore;
     // External contracts
     externalOUSD: Contract;
     externalUSDT: Contract;
     external3CRV: Contract;
 }
 
-export async function buildContractTestContext (): Promise<ContractTestContext> {
+export const signers = ethers.getSigners();
+
+export async function buildContractTestContext (contractRoles: ContractRoles = {}): Promise<ContractTestContext> {
     const context = {} as ContractTestContext;
 
-    [context.owner, context.addr1, context.addr2, context.treasurySigner, context.addr3] = await ethers.getSigners();
-
-    const contracts = await getContractFactories([
-        "CDPosition",
-        "Exchanger",
-        "LeverageEngine",
-        "LeverageAllocator",
-        "PositionToken",
-        "ParameterStore",
-        "VaultOUSD",
-        "LvUSDToken",
-        "Coordinator",
-    ]);
```

Comment:
We are going from getting the factories and deploying separately to doing both concurrently. Main motivation was to consistently provide the admin address as the first argument to each `deploy` but also condenses the code.